bootstrap/pkg/utils/k8utils.go

```diff
@@ -325,3 +146,198 @@ func DeleteResourceFromFile(config *rest.Config, filename string) error {
 	}
 	return nil
 }
+
+type Apply struct {
+	matchVersionKubeConfigFlags *cmdutil.MatchVersionFlags
+	factory                     cmdutil.Factory
+	clientset                   *kubernetes.Clientset
+	options                     *kubectlapply.ApplyOptions
+	tmpfile                     *os.File
+	stdin                       *os.File
+}
+
+func NewApply(kfDef *kfdefs.KfDef) (*Apply, error) {
+	apply := &Apply{
+		matchVersionKubeConfigFlags: cmdutil.NewMatchVersionFlags(genericclioptions.NewConfigFlags()),
+	}
+	apply.factory = cmdutil.NewFactory(apply.matchVersionKubeConfigFlags)
+	clientset, err := apply.factory.KubernetesClientSet()
+	if err != nil {
+		return nil, &kfapis.KfError{
+			Code:    int(kfapis.INTERNAL_ERROR),
+			Message: fmt.Sprintf("could not get clientset Error %v", err),
+		}
+	}
+	apply.clientset = clientset
+	err = apply.namespace(kfDef)
+	if err != nil {
+		return nil, err
+	}
+	return apply, nil
+}
+
+func (a *Apply) DefaultProfileNamespace(name string) bool {
```

Comment:
It seems weird to have a `DefaultProfileNamespace` method for an apply helper struct.
I feel like this is something that should happen inside the kustomize.go file.
How about exposing a `GetClientset` function, so that we can just get a K8s clientset and use it with our client of choice (client-go, controller-runtime).